clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -117,6 +117,24 @@ static void printOmittedTerminatorRegion(mlir::OpAsmPrinter &printer,
                       /*printBlockTerminators=*/!omitRegionTerm(region));
 }
 
+//===----------------------------------------------------------------------===//
+// AllocaOp
+//===----------------------------------------------------------------------===//
+
+void cir::AllocaOp::build(::mlir::OpBuilder &odsBuilder,
```

Comment:
Why the global scope operators here?  Is this a different `mlir` namespace than elsewhere? And `llvm`?

Reply:
MLIR is so many levels deep in templates and auto-generated code here that it makes my head hurt. This function gets called from the call to `create<cir::AllocaOp>` in CIRBaseBuilder.h. This is a static member function in the `cir::AllocaOp` class that gets generated from CIROps.td, but only the declaration is generated, so we have to provide the definition here.

Reply:
Oh! I misunderstood your original comment. This appears to be an MLIR style thing. Maybe @bcardosolopes or @lanza can provide more insight into the history of this. I don't think it's necessary.

Reply:
I don't think this is necessary either - we went back n forth with namespaces in the past couple years, looks like an oversight of some previous bulk change at this point.